CoreLayersView/src/au/gov/asd/tac/constellation/views/layers/components/LayerTitlePane.java

```diff
@@ -252,9 +252,8 @@ public void setValidity(final boolean isValid) {
     }
     
     public void recheckValidity() {
-        boolean validity = query.getQueryString() == null || ExpressionUtilities.testQueryValidity(query.getQueryString());
+        final boolean validity = query.getQueryString() == null || ExpressionUtilities.testQueryValidity(query.getQueryString());
         this.isValid = validity;
-        setValidity(validity);
```

Comment:
Did you need to the layer re-colour to occur? If so, you probably want to keep the function call and remove the `this.isValid...` line instead